ui-speedspacechart/src/components/helpers/drawElements/speedLimitTags.ts

```diff
@@ -43,11 +43,17 @@ const TEXT_PADDING_TOP = 1;
 const ICON_BACKGROUND_WIDTH = 24;
 const ICON_BACKGROUND_HEIGHT = 24;
 
-export const drawSpeedLimitTags = async ({
+export const loadSpeedLimitTagsImages = async () => ({
+  questionImage: await loadSvgImage(questionBlobUrl),
+  alertFillImage: await loadSvgImage(alertFillBlobUrl),
+});
```

Comment:
Hm, I think loading these as a global variable will cause the images to get loaded at startup for any JS webapp which uses the speed limit tags somewhere in their code. For instance, the STDCM OSRD app would load them at startup even though these images are never used there.

Instead we can do something like this perhaps?

https://github.com/OpenRailAssociation/osrd-ui/blob/b1b380e749f5aeb72dd33e081d34a8d978fd3f26/ui-spacetimechart/src/components/WorkScheduleLayer.tsx#L18

Reply:
The code you pointed to is only the export of the helper function, not the actual image loading, which is done by the component in a useeffect with a usestate.

The resulting structure is not too different from the code snippet you shared (we have the image instead of the imageurl as a dependency here, hence we need to have an additional check that the image hasn't been loaded yet, but that's the only difference I think).

Am I missing something?

Reply:
Though moving the last remnants of the fetching logic from speedlimitags to speedlimittagslayer would probably make the code cleaner and easier to follow

Reply:
Ehh, I missed the `() =>` and thought the `await` was executed when loading the JS file, but this is just a function. Sorry about that!